Formalizing customer support tickets process with SET (#467)
mattermost/mattermost-developer-documentation: site/content/internal/sustained-engineering.md

During non-release weeks, triage is held twice a week. Near the end of the release cycle, triage is held daily.

## Customer and Pre-Sales Support Escalation Process

Part of SET's responsibility is to interface with the customer support team. SET's primary goal in respect to support is to triage escalated support issues, work on any high priority customer bugs requiring code change, as well as provide training and knowledge share with the customer support team. When the pre-sales and customer support teams require engineering support, they should:

1. Post the issue in [Sustained Engineering](https://community.mattermost.com/core/channels/sustained-engineering) channel. If the message contains confidential information about a customer, then post in [Customer Support](https://community.mattermost.com/private-core/channels/community) channel in private Staff team.

> 2. At-mention SET team lead, identified as the first person in the [Sustained Engineering](https://community.mattermost.com/core/channels/sustained-engineering) channel header.

Reviewer: ```suggestion
2. At-mention SET team lead, identified as the first person in the [Sustained Engineering](https://community.mattermost.com/core/channels/sustained-engineering) channel header, and provide the relevant information listed in the [support handbook](https://docs.mattermost.com/process/support.html).
```

> 3. If no prompt solution is presented, create a ticket in [Mattermost Jira project](https://mattermost.atlassian.net/) and assign "Sustained Engineering" as the **Mattermost Team**. The ticket will be reviewed by a SET team during Tuesday or Thursday triage.

Reviewer: ```suggestion
3. If no prompt solution is presented, create a ticket in [Mattermost Jira project](https://mattermost.atlassian.net/) and assign "Sustained Engineering" as the **Mattermost Team**. The ticket will be reviewed by SET team during Tuesday or Thursday triage.
```

Reviewer 2: Still think that they should be present in the meeting when the ticket is going to get triaged

Reviewer: FYI that an SA team member tried to join triage meetings this week, but on Tuesday the call was occupied by release discussions (and they had to hop out after 30 mins), on Thursday the Zoom meeting they joined was empty / no one was there.

Not saying we shouldn't have the members join the call, but propose we make it clear how support/pre-sales team participates, and what the fallback is if they can't join (SAs are often travelling to customer sites)

Author: @enahum 

I'm sure if support wants urgent attention they themselves will attend triage and provide feedback. As @jasonblais points out, it might become difficult for them to attend sometimes, and no point in adding something to the process if it's not given any weight.

Lets proceed as is and see how things work out. We can always refine this process further by encouraging them to attend at any time if we deem it necessary :)

Reviewer: ```suggestion
If no solution or next steps (such as a Jira ticket) are presented within 24 hours, create a ticket in [Mattermost Jira project](https://mattermost.atlassian.net/) and assign "Sustained Engineering" as the **Mattermost Team**. The ticket will be reviewed by a SET team during Tuesday or Thursday triage. If you need a resolution sooner than triage time, escalate to @chris.overton on https://community.mattermost.com.
```

> 3. If no prompt solution is presented, create a ticket in [Mattermost Jira project](https://mattermost.atlassian.net/) and assign "Sustained Engineering" as the **Mattermost Team**. The ticket will be reviewed by a SET team member during Tuesday or Thursday triage.

Reviewer 2: Reviewed by the SET team not a member. Also I would recommend that the person that file the ticket to be present at the meeting before the ticket gets triaged to be able to answer any questions that cannot be answered with the info on the ticket

Author: Updated to just `SET team`.

I think any questions that need to be answered regarding the ticket should be done through jira/MM. If SET team thinks they need a high bandwidth conversation with the support team, I suggest that SET engages the support team on the side, as opposed to always getting them to join triage meetings. If something is high priority they'll automatically bring it up during triage meetings.

Thoughts?

> ## Support Responses

Reviewer: ```suggestion
## Customer and Pre-Sales Escalation Process
```

> Part of SET's responsibility is to interface with the customer support team. SET's primary goal in respect to support is to triage escalated support issues, work on any high priority customer bugs requiring code change, as well as provide training and knowledge share with the customer support team. When customer support team requires enginerring support, they should:

Reviewer 2: Solution architects for escalations too, not only support

Author: @enahum do you mean the initial reference to customer support do you mean everywhere we have customer support we should also mention solution architects?

> Part of SET's responsibility is to interface with the customer support team. SET's primary goal in respect to support is to triage escalated support issues, work on any high priority customer bugs requiring code change, as well as provide training and knowledge share with the customer support team. When customer support team requires enginerring support, they should:

Reviewer: ```suggestion
Part of SET's responsibility is to interface with the customer support team. SET's primary goal in respect to support is to triage escalated support issues, work on any high priority customer bugs requiring code change, as well as provide training and knowledge share with the customer support team. When customer pre-sales support team looks for engineering support, they should:
```

Reviewer: ^I think Elias means both support and solution architects can use this process as above

Reviewer 3: Suggestion for final sentence: When the pre-sales customer support team requires engineering support, they should:

Reviewer: Thanks! Updated per suggestion

> 1. Posts the issue on [Customer Support](https://community.mattermost.com/private-core/channels/community) / [SET](https://community.mattermost.com/core/channels/sustained-engineering) channel

Reviewer: ```suggestion
1. Post the issue in [Sustained Engineering](https://community.mattermost.com/core/channels/sustained-engineering) channel. If the message contains confidential information about a customer, then post in [Customer Support](https://community.mattermost.com/private-core/channels/community) channel in private Staff team.
```

> 2. Engage someone from SET to provide an initial assement

Reviewer: ```suggestion
2. At-mention SET team lead, identified as the first person in the [Sustained Engineering](https://community.mattermost.com/core/channels/sustained-engineering) channel header.
```

Reviewer: ^propose we specify how and who to engage (at-mention, SET team lead)

> 3. If no prompt solution is presented, a ticket should be created by customer support and assigned to SET for further investigation

Reviewer: ```suggestion
3. If no prompt solution is presented, create a ticket in [Mattermost Jira project](https://mattermost.atlassian.net/) and assign "Sustained Engineering" as the **Mattermost Team**. The ticket will be reviewed by a SET team member during Tuesday or Thursday triage.
```

> This process helps increase accountability and tracibility for both teams.

Reviewer: What do you mean by "both teams"?

Author: Engineering and customer support. Maybe I should remove `for both teams?`

Reviewer: Got it, perhaps something like:

```suggestion
This process increases accountability and traceability across Mattermost.
```

Reviewer: Sorry, I missed this note @ali-farooq0. How about

```suggestion
This process helps increase accountability and traceability across teams.
```

Author: ah good catch, missed that too. Updated.

> 

Reviewer: ### Support Response Templates

Author: Updated.

> SET members should use the following responses when receiving issue escalations from the customer support team:

Reviewer: ```suggestion
SET members should use the following responses when receiving issue escalations from the customer or pre-sales support teams:
```
